### PR TITLE
Don't import Microsoft.WinFX.targets for sdk style projects

### PR DESCRIPTION
**Pull request by BenVillalobos**
Part 1 to fix https://github.com/microsoft/msbuild/issues/4948

Part 2 (sets the ImportWinFXTargets property to false): https://github.com/dotnet/sdk/pull/10998

This conditionally imports winfx.targets for all non-sdk style builds. Pairs with another PR for the sdk 
that ensures the condition will fall through.

### CODE REVIEW

File: src/Tasks/Microsoft.WinFx.targets [hunk 1]
```diff
@@ -9,6 +9,6 @@
         the shim will internally simply redirect to the real copy of the targets file in the .NET Framework. 
    -->
 
-   <Import Project="$(MSBuildFrameworkToolsPath)\Microsoft.WinFx.targets" Condition="Exists('$(MSBuildFrameworkToolsPath)\Microsoft.WinFx.targets')" />
+   <Import Project="$(MSBuildFrameworkToolsPath)\Microsoft.WinFx.targets" Condition="Exists('$(MSBuildFrameworkToolsPath)\Microsoft.WinFx.targets') and '$(ImportWinFXTargets)' != 'false'" />
```
Comment (cdmihai) [inline src/Tasks/Microsoft.WinFx.targets]: Check the property first to avoid doing the IO existence check. The and operator is short circuiting.

Comment (BenVillalobos) [reply src/Tasks/Microsoft.WinFx.targets]: Good call, done!

File: src/Tasks/Microsoft.WinFx.targets [hunk 2]
```diff
@@ -9,6 +9,6 @@
         the shim will internally simply redirect to the real copy of the targets file in the .NET Framework. 
    -->
 
-   <Import Project="$(MSBuildFrameworkToolsPath)\Microsoft.WinFx.targets" Condition="Exists('$(MSBuildFrameworkToolsPath)\Microsoft.WinFx.targets')" />
+   <Import Project="$(MSBuildFrameworkToolsPath)\Microsoft.WinFx.targets" Condition="'$(ImportWinFXTargets)' != 'false' and Exists('$(MSBuildFrameworkToolsPath)\Microsoft.WinFx.targets')" />
```
Comment (Nirmal4G) [inline src/Tasks/Microsoft.WinFx.targets]: Why not use `ImportInboxWinFXTargets` or `ImportFrameworkWinFXTargets`, seems more clear as to what we're not importing and where it's from. I prefer the former name.

Comment (BenVillalobos) [reply src/Tasks/Microsoft.WinFx.targets]: Good suggestion. I opted for ImportFrameworkWinFXTargets. 

Thanks!

**Comment by BenVillalobos** [comment]
Verified that this works locally 👍

File: src/Shared/Traits.cs [hunk 3]
```diff
@@ -109,6 +109,8 @@ internal class EscapeHatches
         /// </summary>
         public readonly bool AlwaysUseContentTimestamp = Environment.GetEnvironmentVariable("MSBUILDALWAYSCHECKCONTENTTIMESTAMP") == "1";
 
+        public readonly bool LogTaskInputsVerbose = Environment.GetEnvironmentVariable("MSBUILDLOGTASKINPUTSVERBOSE") == "1";
```
Comment (rainersigwald) [inline src/Shared/Traits.cs:112]: Did this leak in from your other change?

Comment (BenVillalobos) [reply src/Shared/Traits.cs:112]: Whoops! I'll revert that.

Thanks!

**Pull request merged**